common/test/views/support/cleaner/AmpEmbedCleanerTest.scala

```diff
@@ -64,6 +80,34 @@ class AmpEmbedCleanerTest extends FlatSpec with Matchers {
     clean(document)
   }
 
+
+  private def cleanDocumentWithCommentEmbed(): Document = {
+
+    val doc = <html>
+      <body>
+      <figure class="element element-comment" data-canonical-url="https://discussion.theguardian.com/comment-permalink/88222201">
+        <div class="d2-comment-embedded" itemtype="http://schema.org/Comment">
+          <div class="d2-left-col">
+            <a href="https://profile.theguardian.com/user/id/12345678">
+              <img class="d2-avatar" src="https://avatar.guim.co.uk/user/15301515" height="40" width="40" alt="User avatar for fooBar"/>
+            </a>
+          </div>
+          <div class="d2-right-col">
+          </div>
+          <div class="d2-permalink">
+          </div>
+          <div class="d2-body" itemprop="text">
+            <p>foo bar foo foo bar</p>
+          </div>
+        </div>
+      </figure>
+      </body>
+      </html>.toString()
+    val document = Jsoup.parse(StringEscapeUtils.unescapeXml(doc))
```

Comment:
It's parsing the doc as XML, so when you do "toString()" on it, it ads a whole lot of XML code in to the string (ie '<' becomes '&lt;'). This then caused problems with Jsoup, when it was parsed.
Maybe just convert all the test data back to strings. The display is noce and readable this way, but on reflection, it is a bit weird having to convert it back from XML.